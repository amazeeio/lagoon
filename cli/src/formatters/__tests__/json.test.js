// @flow

import formatAsJson from '../json';

describe('json formatter', () => {
  it('should format data with 0 data rows correctly', () => {
    const formatted = formatAsJson([['Name', 'Customer']]);
    expect(formatted).toMatchSnapshot();
  });

  it('should format data with 1 data row correctly', () => {
    const formatted = formatAsJson([
      ['Name', 'Customer'],
      ['test-project-json-1', 'test-customer-json-1'],
    ]);
    expect(formatted).toMatchSnapshot();
  });

  it('should format data with > 1 data rows correctly', () => {
    const formatted = formatAsJson([
      ['Name', 'Customer'],
      ['test-project-json-1', 'test-customer-json-1'],
      ['test-project-json-2', 'test-customer-json-2'],
    ]);
    expect(formatted).toMatchSnapshot();
  });
});
