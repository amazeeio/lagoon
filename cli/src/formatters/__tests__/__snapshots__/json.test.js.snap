// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`json formatter should format data with > 1 data rows correctly 1`] = `
Array [
  Array [
    "Name",
    "Customer",
  ],
  Array [
    "test-project-json-1",
    "test-customer-json-1",
  ],
  Array [
    "test-project-json-2",
    "test-customer-json-2",
  ],
]
`;

exports[`json formatter should format data with 0 data rows correctly 1`] = `
Array [
  Array [
    "Name",
    "Customer",
  ],
]
`;

exports[`json formatter should format data with 1 data row correctly 1`] = `
Array [
  Array [
    "Name",
    "Customer",
  ],
  Array [
    "test-project-json-1",
    "test-customer-json-1",
  ],
]
`;
